### PR TITLE
Unify networking benchmarks

### PR DESCRIPTION
**Pull request by rsevilla87**
### Description

Join networking benchmarks into a single script ./run.sh and improve README.md

This is a breaking change, we should push changes to airflow (or use a specific e2e-benchmarking commit/version) before merging it.

### CODE REVIEW

File: workloads/network-perf/README.md [hunk 1]
```diff
@@ -1,211 +1,69 @@
...
-1. pod to pod using Hostnetwork
-2. pod to pod using SDN
-3. pod to pod using Service
-4. pod to pod using Multus (NetworkAttachmentDefinition needs to be provided)
+There are 5 types network tests:
+
+1. pod to pod using SDN: `WORKLOAD=pod2pod ./run.sh`
+2. pod to pod using SDN and network policy: `WORKLOAD=pod2pod NETWORK_POLICY=true ./run.sh`
+3. pod to pod using Hostnetwork: `WORKLOAD=pod2pod HOSTNETWORK=true ./run.sh`
+4. pod to service: `WORKLOAD=pod2svc ./run.sh`
```
Comment (dry923) [inline workloads/network-perf/README.md:9]: There was also the service ip one with network policy ```run_serviceip_network_policy_test_fromgit.sh```

File: workloads/network-perf/README.md [hunk 2]
```diff
@@ -1,211 +1,69 @@
...
+| **ADDRESSPOOL**         | To provide MetalLB addresspool for a service, this will be used as LoadBalancer network. Mentioned addresspool should be pre-provisioned before execution of this script. | addresspool-l2 |
+| **SERVICE_ETP**         | To mention the type of `ExternalTrafficPolicy` of a service, supported option `Cluster` or `Local` | Cluster |
+| **SAMPLES**             | How many times to run the tests | 3 |
+| **MULTI_AZ**            | If true, uperf client and server pods will be colocated in different topology zones or AZs (`topology.kubernetes.io/zone` in k8s terminology). 2 worker nodes in differet topology zones are required to enable this flag.  If false, uperf client and server pods will be colocated in the same topology zone | true |
+| **PAIRS**               | List with the number of pairs the test will be triggered | 1 2 4 |
+| **TEST_TIMEOUT**        | Benchmark timeout, in seconds | 7200 (2 hours) |
+| **TEST_CLEANUP**        | Remove benchmark CR at the end | true | 
+| **CERBERUS_URL**        | URL to check the health of the cluster using Cerberus | "" |
```
Comment (dry923) [inline workloads/network-perf/README.md:39]: Can we remove Cerberus since we haven't ever really used it?

Comment (rsevilla87) [reply workloads/network-perf/README.md:39]: I was tempted, but... it's not out of scope of this PR? :)

File: workloads/network-perf/run.sh [hunk 3]
```diff
@@ -1,20 +1,53 @@
...
+    export SERVICEIP=true
+  ;;
+  hostnet)
     export HOSTNETWORK=true
-elif [[ $WORKLOAD == "service" ]]; then
-    export SERVICEIP=true 
+  ;;
+  smoke)
+    CR=smoke-crd.yaml
+    export SAMPLES=1
```
Comment (dry923) [inline workloads/network-perf/run.sh:19]: We should reduce the number of pairs for the smoke test as well. No need to do all 3 IMHO

File: workloads/network-perf/README.md (hunk 1 above)
Comment (rsevilla87) [reply workloads/network-perf/README.md:9]: Added!

File: workloads/network-perf/README.md (hunk 2 above)
Comment (dry923) [reply workloads/network-perf/README.md:40]: I mean if you are doing a full re-write I'd say its in scope but i'll let the others chime in as well.

File: workloads/network-perf/run.sh (hunk 3 above)
Comment (dry923) [reply workloads/network-perf/run.sh:18]: ack. figured we could save a few seconds but s'all good.

File: workloads/network-perf/smoke-crd.yaml [hunk 4]
```diff
@@ -1,9 +1,8 @@
 ---
-
 apiVersion: ripsaw.cloudbulldozer.io/v1alpha1
 kind: Benchmark
 metadata:
-  name: uperf-benchmark-${WORKLOAD}-network-${pairs}
+  name: uperf-${WORKLOAD}-network-${PAIRS}
```
Comment (dry923) [inline workloads/network-perf/smoke-crd.yaml:5]: Really a minor thought but shouldn't this be something like ```uperf-${WORKLOAD}-pairs-${PAIRS}``` ?
The word network in there is kind of redundant with uperf and at a glance the last number doesn't really say anything as it is today.

Comment (rsevilla87) [reply workloads/network-perf/smoke-crd.yaml]: fixed!

File: workloads/network-perf/run.sh (hunk 3 above)
Comment (rsevilla87) [reply workloads/network-perf/run.sh:18]: Runtime has been 00:05:52, not that bad :)

Comment (dry923) [reply workloads/network-perf/run.sh:19]: ack. WFM

**Review by mkarg75 — APPROVED** [review]
lgtm!

**Review by morenod — APPROVED** [review]
LGTM

**Review by Harshith-umesh — COMMENTED** [review]
LGTM but as you've said we should first cut a release of e2e, and make changes in airflow to handle this.

**Comment by morenod** [comment]
This PR on Airflow is aligned with this change
https://github.com/cloud-bulldozer/airflow-kubernetes/pull/132

File: workloads/network-perf/README.md (hunk 2 above)
Comment (rsevilla87) [reply workloads/network-perf/README.md]: Added a new commit to remove it ;)

**Pull request merged**